corehq/util/datadog/lockmeter.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import time
+
+from corehq.util.datadog.gauges import datadog_counter, datadog_bucket_timer
+
+
+class LockMeter(object):
+    """A lock wrapper that measures various lock characteristics
+
+    This was built for profiling Redis locks, but will for any type of
+    lock that has `acquire()` and `release()` methods.
+    """
+
+    timing_buckets = (0.1, 1, 5, 10, 30, 60, 120, 60 * 5, 60 * 10, 60 * 15, 60 * 30)
+
+    def __init__(self, lock, name, track_unreleased=True):
```

Comment:
Nit: The instantiation followed a really similar pattern each time of creating a redis lock and then passing that in; I might flatten that to a single helper function `get_metered_redis_lock(key, timeout, name, track_unreleased)`. Not sure if you like that pattern or not

Reply:
https://github.com/dimagi/commcare-hq/pull/23428/commits/ff3eac98c5c293ef8365f40b4a6ca038641abff6

Good idea. Makes the imports cleaner too. The one issue with this is that the `key` parameter is actually our convention (which I continued), and the redis parameter where `key` is passed is actually called `name`. I added a docstring to hopefully resolve the ambiguity between that and the other (new) parameter called `name` used to populate the datadog tag.

Reply:
Cool, I like the way that cleaned up the imports and the calling code